listenbrainz/webserver/static/js/src/FollowButton.tsx

```diff
@@ -0,0 +1,137 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+import * as React from "react";
+import APIService from "./APIService";
+
+type FollowButtonProps = {
+  user: ListenBrainzUser;
+  loggedInUser: ListenBrainzUser;
+  loggedInUserFollowsUser: boolean;
+};
+
+type FollowButtonState = {
+  loggedInUserFollowsUser: boolean;
+  justFollowed: boolean;
+  hover: boolean;
+  error: boolean;
+};
+
+class FollowButton extends React.Component<
+  FollowButtonProps,
+  FollowButtonState
+> {
+  APIService: APIService;
+
+  constructor(props: FollowButtonProps) {
+    super(props);
+    this.state = {
+      loggedInUserFollowsUser: props.loggedInUserFollowsUser,
+      hover: false,
+      justFollowed: false,
+      error: false,
+    };
+
+    this.APIService = new APIService(`${window.location.origin}/1`);
+  }
+
+  toggleHover = () => {
+    const { hover } = this.state;
+    this.setState({ hover: !hover, justFollowed: false });
+  };
+
+  handleButtonClick = () => {
+    const { loggedInUserFollowsUser } = this.state;
+    if (loggedInUserFollowsUser) {
+      this.unfollowUser();
+    } else {
+      this.followUser();
+    }
+  };
+
+  followUser = () => {
+    const { user } = this.props;
+    this.APIService.followUser(user.name).then(({ status }) => {
+      if (status === 200) {
+        this.setState({ loggedInUserFollowsUser: true, justFollowed: true });
+      } else {
+        this.setState({ error: true });
+      }
+    });
+  };
+
+  unfollowUser = () => {
+    const { user } = this.props;
+    this.APIService.unfollowUser(user.name).then(({ status }) => {
+      if (status === 200) {
+        this.setState({ loggedInUserFollowsUser: false, justFollowed: false });
+      } else {
+        this.setState({ error: true });
+      }
+    });
+  };
+
+  getButtonDetails = (): { buttonClass: string; buttonText: string } => {
+    const { error, justFollowed, loggedInUserFollowsUser, hover } = this.state;
+
+    if (error) {
+      return {
+        buttonClass: "btn btn-sm btn-danger",
+        buttonText: "Something went wrong!",
+      };
+    }
+
+    if (justFollowed) {
+      return { buttonClass: "btn btn-sm btn-info", buttonText: "Following" };
+    }
+
+    if (loggedInUserFollowsUser) {
+      if (!hover) {
+        return { buttonClass: "btn btn-sm btn-info", buttonText: "Following" };
```

Comment:
`btn-secondary` instead of `btn-info`?

Reply:
Is that the same as `btn-success` (green)?
I would expect green if everything is OK